### PR TITLE
[backend] Auth | Init data | create admin account

### LINKED ISSUE #106: [backend] boot script add admin account
**Describe the solution you'd like**
run query check if the admin account exit else add admin account

### PR DESCRIPTION
**Pull request by x06lan**
Ex. Resolves #106 #107 #127

### Changes

* when backend sever start will add admin account if did not have any admin account in database 
* script add and remove test data (will add or delete image file in minio as well)
* add auth middleware to identify username and role
* fix shop report time implicitly convert

> [!CAUTION]  
> After merge this PR most API( beside `general` API) will reponse `401` If not have auth token in header

### Testing Approach (if unconventional)

> [!IMPORTANT]  
> add this to .env
> ```
> TWP_ADMIN_USER=admin 
> TWP_ADMIN_PASSWORD=Secret@123
> ```
> download this [data.json](https://gist.github.com/x06lan/b233be13dca733a2eee96d1ef8a8ab20)
> new data.json load image by file path(image in ./frontend/public/placeholder/) instead of use some random UUID
> 

```zsh
# add this code to main.go insert Data
go run script/main.go load data.json
# add this code to main.go delete Data
go run script/main.go unload data.json
```
due to minio issues currently only localhost can access image by presigned url 
```bash
docker-compose up postgres minio 
npm run build 
go run ./
```
check have admin account in database
screenshot-2023-12-28-01-41-39
check have load data image url are correct like product and user image

### CODE REVIEW

File: db/queries/admin.sql
Comment (Error0229) [inline db/queries/admin.sql]: syntax error

Comment (NOOBDY) [inline db/queries/admin.sql]: is this a typo

File: main.go
Comment (NOOBDY) [inline main.go]: does this need to be here

File: test/db/main.go
Comment (Error0229) [inline test/db/main.go:1]: Is this file necessary to exist?

**Review by kryr34 — CHANGES_REQUESTED** [review]
plz remove reformatted SQL strings

File: pkg/script/main.go
Comment (NOOBDY) [inline pkg/script/main.go:1]: make this into a separate executable

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: go.mod
Comment (NOOBDY) [inline go.mod:1]: run `go mod tidy` here

File: test/db/main.go
Comment (NOOBDY) [inline test/db/main.go:1]: why delete this

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: test/db/main.go
Comment (x06lan) [reply test/db/main.go:1]: since `go run script/main.go load data.json` can test db load data there is no need to keep this file

Comment (NOOBDY) [reply test/db/main.go:1]: this file is for testing if inserting data works, it serves a different purpose

**Review by NOOBDY — APPROVED** [review]

**Review by Error0229 — APPROVED** [review]
WOMM

**Review by ntut-rick — COMMENTED** [review]
uh

**Review by kryr34 — COMMENTED** [review]
Dismiss change request

**Review by kryr34 — APPROVED** [review]
Dismiss change request

**Comment by UGisBusy** [comment]
login with admin and passing auth header get 403 for many apis
i added this thing when fetch:
```
Authorization: `Bearer ${token}`,
```

**Review by KatoToyoki — APPROVED** [review]
backend WOMM while frontend is broken, we will fix after it merged

**Pull request merged**
